runtime/compiler/env/PersistentAllocator.cpp

```diff
@@ -164,26 +225,34 @@ PersistentAllocator::remainingSpace(J9MemorySegment &segment) throw()
    }
 
 void
-PersistentAllocator::freeBlock(Block * block)
+PersistentAllocator::freeFixedSizeBlock(Block * block)
    {
+   // Appropriate lock should have been obtained
    TR_ASSERT(block->_size > 0, "Block size is non-positive");
-   TR_ASSERT(block->_next == NULL, "In-use persistent memory block @ belongs to a free block chain.", block);
-   block->_next = NULL;
+   size_t const index = freeBlocksIndex(block->_size);
+   TR_ASSERT(index != LARGE_BLOCK_LIST_INDEX, "freeFixedSizeBlock should be used for small blocks, so index cannot be LARGE_BLOCK_LIST_INDEX");
+   block->_next = _freeBlocks[index];
+   _freeBlocks[index] = block;
+   }
 
-   // If this is a small block, add it to the appropriate fixed-size-block
-   // chain. Otherwise add it to the variable-size-block chain which is in
-   // ascending size order.
+void
+PersistentAllocator::freeVariableSizeBlock(Block * block)
+   {
+   // Appropriate lock should have been obtained
+   TR_ASSERT(block->_size > 0, "Block size is non-positive");
+   block->_next = NULL;
+   Block * volatile next = block->_next;
```

Comment:
This line was added as part of your last update; doesn't seem to be used.

Reply:
Good catch! This is me trying to determine the effect/correctness of volatile in connection to the other defect involving j9method->extra.